File: 1444-number-of-steps-to-reduce-a-number-to-zero/number-of-steps-to-reduce-a-number-to-zero.cpp
class Solution {
public:
    int numberOfSteps(int num) {
         int step_count = 0; // snake_case variable
        
        // Jab tak number 0 nahi hota, tab tak loop chalega
        while (num > 0) {
            // % 2 == 0 matlab even check kar rahe hain
            if (num % 2 == 0) {
                num /= 2; // num = num / 2 likhne ka short tarika
            } 
            else {
                num -= 1; // num = num - 1 likhne ka short tarika
            }
            
            step_count++; // Step badha diya
        }
        
        return step_count;
    }
};

File: 1444-number-of-steps-to-reduce-a-number-to-zero/number-of-steps-to-reduce-a-number-to-zero.java
class Solution {
    public int numberOfSteps(int num) {
          int stepCount = 0; // camelCase variable
        
        // Jab tak number 0 nahi hota, loop chalta rahega
        while (num > 0) {
            // Agar number even hai (2 se poora divide ho raha hai)
            if (num % 2 == 0) {
                num = num / 2;
            } 
            // Agar number odd hai
            else {
                num = num - 1;
            }
            
            // Har baar ek step badha do
            stepCount++;
        }
        
        return stepCount;
    }
}